Author: Giuseppe Iuculano <giuseppe@iuculano.it>
Description: Some little fixes to silence compiler.
--- a/chklastlog.c
+++ b/chklastlog.c
@@ -173,7 +173,7 @@ int main(int argc, char*argv[]) {
             {
                 if (*uid > MAX_ID)
                 {
-                   fprintf(stderr, "MAX_ID is %ld and current uid is %ld, please check\n\r", MAX_ID, *uid );
+                   fprintf(stderr, "MAX_ID is %ld and current uid is %ld, please check\n\r", (long int)MAX_ID, (long int)*uid );
                    exit (1);
 
                 }
--- a/chkwtmp.c
+++ b/chkwtmp.c
@@ -30,6 +30,7 @@
 #ifdef SOLARIS2
 #include <fcntl.h>
 #endif
+#include <stdlib.h>
 
 #ifdef __FreeBSD__
 #define WTMP_FILENAME "/var/log/wtmp"
