Description: Custom configuration options for Debian package
Author: Julien Valroff <julien@debian.org>
Forwarded: not-needed

--- a/files/rkhunter.conf
+++ b/files/rkhunter.conf
@@ -99,17 +99,17 @@
 # important files will be written to this directory, so be
 # sure that the directory permissions are tight.
 #
-#TMPDIR=/var/lib/rkhunter/tmp
+TMPDIR=/var/lib/rkhunter/tmp
 
 #
 # Specify the database directory to use.
 #
-#DBDIR=/var/lib/rkhunter/db
+DBDIR=/var/lib/rkhunter/db
 
 #
 # Specify the script directory to use.
 #
-#SCRIPTDIR=/usr/local/lib/rkhunter/scripts
+SCRIPTDIR=/usr/share/rkhunter/scripts
 
 #
 # This option can be used to modify the command directory list used
@@ -255,8 +255,15 @@
 # Please read the README file for more details about enabling and disabling
 # tests, the test names, and how rkhunter behaves when these options are used.
 #
+# hidden_procs test requires the unhide command which is part of the unhide
+# package in Debian.
+#
+# apps test is disabled by default as it triggers warnings about outdated 
+# applications (and warns about possible security risk: we better trust
+# the Debian Security Team).
+#
 ENABLE_TESTS="all"
-DISABLE_TESTS="suspscan hidden_ports hidden_procs deleted_files packet_cap_apps"
+DISABLE_TESTS="suspscan hidden_procs deleted_files packet_cap_apps apps"
 
 #
 # The HASH_FUNC option can be used to specify the command to use
@@ -324,6 +331,9 @@
 #
 # Whenever this option is changed 'rkhunter --propupd' must be run.
 #
+# NONE is the default for Debian as well, as running --propupd takes
+# about 4 times longer when it's set to DPKG
+#
 #PKGMGR=NONE
 
 #
@@ -466,8 +476,14 @@
 # be specified more than once. The option may use wildcard
 # characters.
 #
-#SCRIPTWHITELIST="/sbin/ifup /sbin/ifdown"
-#SCRIPTWHITELIST="/usr/bin/groups"
+SCRIPTWHITELIST=/bin/egrep
+SCRIPTWHITELIST=/bin/fgrep
+SCRIPTWHITELIST=/bin/which
+SCRIPTWHITELIST=/usr/bin/groups
+SCRIPTWHITELIST=/usr/bin/ldd
+SCRIPTWHITELIST=/usr/bin/lwp-request
+SCRIPTWHITELIST=/usr/sbin/adduser
+SCRIPTWHITELIST=/usr/sbin/prelink
 
 #
 # Allow the specified commands to have the immutable attribute set.
@@ -492,11 +508,9 @@
 # may use wildcard characters.
 #
 #ALLOWHIDDENDIR="/etc/.java"
-#ALLOWHIDDENDIR="/dev/.udev /dev/.udevdb /dev/.udev.tdb"
 #ALLOWHIDDENDIR="/dev/.static"
-#ALLOWHIDDENDIR="/dev/.initramfs"
 #ALLOWHIDDENDIR="/dev/.SRC-unix"
-#ALLOWHIDDENDIR="/dev/.mdadm"
+#ALLOWHIDDENDIR="/etc/.etckeeper"
 
 #
 # Allow the specified hidden files to be whitelisted.
@@ -521,6 +535,9 @@
 #ALLOWHIDDENFILE="/usr/lib/hmaccalc/sha384hmac.hmac"
 #ALLOWHIDDENFILE="/usr/lib/hmaccalc/sha512hmac.hmac"
 #ALLOWHIDDENFILE="/usr/sbin/.sshd.hmac"
+#ALLOWHIDDENFILE="/usr/share/man/man5/.k5login.5.gz"
+#ALLOWHIDDENFILE="/etc/.gitignore"
+#ALLOWHIDDENFILE="/etc/.bzrignore"
 
 #
 # Allow the specified processes to use deleted files. The
@@ -535,8 +552,10 @@
 # characters, but only in the file names.
 #
 #ALLOWPROCDELFILE="/sbin/cardmgr /usr/sbin/gpm:/etc/X11/abc"
-#ALLOWPROCDELFILE="/usr/libexec/gconfd-2"
+#ALLOWPROCDELFILE="/usr/lib/libgconf2-4/gconfd-2"
 #ALLOWPROCDELFILE="/usr/sbin/mysqld:/tmp/ib*"
+#ALLOWPROCDELFILE="/usr/lib/iceweasel/firefox-bin"
+#ALLOWPROCDELFILE="/usr/bin/file-roller"
 
 #
 # Allow the specified processes to listen on any network interface.
@@ -547,7 +566,6 @@
 #ALLOWPROCLISTEN="/sbin/dhclient /usr/bin/dhcpcd"
 #ALLOWPROCLISTEN="/usr/sbin/pppoe /usr/sbin/tcpdump"
 #ALLOWPROCLISTEN="/usr/sbin/snort-plain"
-#ALLOWPROCLISTEN="/usr/local/bin/wpa_supplicant"
 
 #
 # Allow the specified network interfaces to be in promiscuous mode.
@@ -647,7 +665,7 @@
 # The option may be specified more than once. The option may use
 # wildcard characters.
 #
-#STARTUP_PATHS="/etc/rc.d /etc/rc.local"
+#STARTUP_PATHS="/etc/init.d /etc/rc.local"
 
 #
 # This setting tells rkhunter the pathname to the file containing the
@@ -668,7 +686,7 @@
 # NOTE: For *BSD systems you will probably need to use this option
 # for the 'toor' account.
 #
-#UID0_ACCOUNTS="toor rooty"
+#UID0_ACCOUNTS="toor rooty sashroot"
 
 #
 # Allow the following accounts to have no password. NIS/YP entries do
@@ -786,7 +804,7 @@
 # specified, then RKH will assume the O/S release information is on the
 # first non-blank line of the file.
 #
-#OS_VERSION_FILE="/etc/release"
+#OS_VERSION_FILE="/etc/debian_version"
 
 #
 # The following two options can be used to whitelist files and directories
@@ -979,4 +997,6 @@
 # of 2 will disable the Ruby 'unhide.rb' program. The default value is 0. To disable
 # both programs, then disable the 'hidden_procs' test.
 #
-#DISABLE_UNHIDE=0
+DISABLE_UNHIDE=1
+
+INSTALLDIR="/usr"
